lib/Doctrine/ORM/Cache/DefaultEntityHydrator.php

```diff
@@ -73,7 +73,7 @@ public function __construct(EntityManagerInterface $em)
     public function buildCacheEntry(ClassMetadata $metadata, EntityCacheKey $key, $entity)
     {
         $data = $this->uow->getOriginalEntityData($entity);
-        $data = array_merge($data, $key->identifier); // why update has no identifier values ?
```

Comment:
There's only one use case where this can happen, which is through "loadById()", where you don't yet have an entity reference to retrieve the information through UnitOfWork.

Deferring to Hydrator indeed solves the problem, but I'd argue this is not the ideal solution, but rather find a common ground in the loadById in specific somehow. Requires further investigation